common/djangoapps/enrollment/views.py

```diff
@@ -581,27 +586,41 @@ def post(self, request):
                 )
 
             enterprise_course_consent = request.data.get('enterprise_course_consent')
-            # Check if the enterprise_course_enrollment is a boolean
-            if has_api_key_permissions and enterprise_enabled() and enterprise_course_consent is not None:
-                if not isinstance(enterprise_course_consent, bool):
-                    return Response(
-                        status=status.HTTP_400_BAD_REQUEST,
-                        data={
-                            'message': (u"'{value}' is an invalid enterprise course consent value.").format(
-                                value=enterprise_course_consent
-                            )
-                        }
-                    )
-                try:
-                    EnterpriseApiClient().post_enterprise_course_enrollment(
-                        username,
-                        unicode(course_id),
-                        enterprise_course_consent
-                    )
-                except EnterpriseApiException as error:
-                    log.exception("An unexpected error occurred while creating the new EnterpriseCourseEnrollment "
-                                  "for user [%s] in course run [%s]", username, course_id)
-                    raise CourseEnrollmentError(error.message)
+            explicit_linked_enterprise = request.data.get('linked_enterprise_customer')
+            if has_api_key_permissions and enterprise_enabled():
+                # We received an explicitly-linked EnterpriseCustomer for the enrollment
+                if explicit_linked_enterprise is not None:
+                    kwargs = {
+                        'username': username,
+                        'course_id': unicode(course_id),
+                        'enterprise_customer_uuid': explicit_linked_enterprise,
+                    }
+                    consent_client = ConsentApiClient()
+                    if consent_client.consent_required(**kwargs):
+                        consent_client.provide_consent(**kwargs)
```

Comment:
We can remove the first API call, and it should generally work as expected. :)

As far as logging, we definitely can. I didn't include any initially because getting an error code from that API is _not_ an expected state at all, and this flow doesn't have any direct user input to mess with things.